pkg/evaluator/evaluator.go

```diff
@@ -16,21 +18,28 @@ func NewEvaluator(builtins Builtins) *Evaluator {
 }
 
 type Evaluator struct {
-	Stopped  bool
-	Yielder  Yielder // Yield to give JavaScript/browser events a chance to run.
-	print    func(string)
-	builtins Builtins
+	Stopped       bool
+	Yielder       Yielder // Yield to give JavaScript/browser events a chance to run.
+	print         func(string)
+	builtins      Builtins
+	eventHandlers map[string]*parser.EventHandler //
 
 	scope *scope // Current top of scope stack
 }
 
+type Event interface {
```

Comment:
Does this interface add much? What if it were:

```
type Event struct {
  Name string
  Params []any
}
```
and instead of the `xyEvent` and `stringEvent` structs, you could just have
```
func newXYEvent(name string, x, y float64) *Event {
  return &Event{Name: name, Params: []any{x, y}}
}
func newStringEvent(name, str string) *Event {
  return &Event{Name:name, Params: []any{str}}
}
```
and the use these functions in the Go handlers?

Because nothing seems to access the individual fields of the `xyEvent` and `stringEvent` structs.

Reply:
ok, agreed.

Reply:
really much better actually. thank you.